controllers/build.go

```diff
@@ -0,0 +1,263 @@
+			{
+				Name: "workspace",
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "appstudio",
+				},
+				SubPath: component.Name + "/" + workspaceSubPath, //$(tt.params.git-revision)/",
+			},
+			{
+				Name: "registry-auth",
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "redhat-appstudio-registry",
+				},
+			},
+		},
+	}
+	return pipelineRunSpec
+}
+
+func normalizeOutputImageURL(outputImage string) string {
+
+	// if provided image format was
+	//
+	// quay.io/foo/bar:mytag , then
+	// push to quay.io/foo/bar:mytag-git-revision
+	//
+	// else,
+	// push to quay.io/foo/bar:git-revision
+
+	if strings.Contains(outputImage, ":") {
+		outputImage = outputImage + "-" + "$(tt.params.git-revision)"
+	} else {
+		outputImage = outputImage + ":" + "$(tt.params.git-revision)"
+	}
+	return outputImage
+}
+
+func paramsForInitialBuild(component appstudiov1alpha1.Component) []tektonapi.Param {
+	sourceCode := component.Spec.Source.GitSource.URL
+	outputImage := component.Spec.Build.ContainerImage
+
+	params := []tektonapi.Param{
+		{
+			Name: "git-url",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: sourceCode,
+			},
+		},
+		{
+			Name: "output-image",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: outputImage,
+			},
+		},
+	}
+
+	return params
+}
+
+func paramsForWebhookBasedBuilds(component appstudiov1alpha1.Component) []tektonapi.Param {
+	sourceCode := component.Spec.Source.GitSource.URL
+	outputImage := normalizeOutputImageURL(component.Spec.Build.ContainerImage)
+
+	params := []tektonapi.Param{
+		{
+			Name: "git-url",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: sourceCode,
+			},
+		},
+		{
+			Name: "output-image",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: outputImage,
+			},
+		},
+	}
+
+	return params
+}
+
+func commonAnnotations() map[string]string {
+	annotations := map[string]string{
+		"build.appstudio.openshift.io/build":   "true",
+		"build.appstudio.openshift.io/type":    "build",
+		"build.appstudio.openshift.io/version": "0.1",
+	}
+	return annotations
+}
+
+func commonStorage(name string, namespace string) corev1.PersistentVolumeClaim {
+	fsMode := corev1.PersistentVolumeFilesystem
+
+	workspaceStorage := &corev1.PersistentVolumeClaim{
+		ObjectMeta: v1.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Annotations: commonAnnotations(),
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				"ReadWriteOnce",
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					"storage": resource.MustParse("10Mi"),
+				},
+			},
+			VolumeMode: &fsMode,
+		},
+	}
+	return *workspaceStorage
+}
+
+func route(component appstudiov1alpha1.Component) routev1.Route {
+	var port int32 = 8080
```

Comment:
should be?
```suggestion
	var port int32 = int32(component.Spec.TargetPort)
```

Reply:
This is not the component's port, rather the one associated with the tekton eventlistener's service.